substrate/frame/multisig/src/lib.rs

```diff
@@ -157,7 +157,28 @@ pub mod pallet {
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: weights::WeightInfo;
 
-		/// Provider for the block number. Normally this is the `frame_system` pallet.
+		/// Query the current block number.
+		///
+		/// Must return monotonically increasing values when called from consecutive blocks.
+		/// Can be configured to return either:
+		/// - the local block number of the runtime via `frame_system::Pallet`
+		/// - a remote block number, eg from the relay chain through `RelaychainDataProvider`
+		/// - an arbitrary value through a custom implementation of the trait
+		///
+		/// There is currently no migration provided to "hot-swap" block number providers and it may
+		/// result in undefined behavior when doing so. Parachains are therefore best of to set this
+		/// to their local block number provider if they have the pallet already deployed.
+		///
+		/// Suggested values:
+		/// - Solo- and Relay-chains: `frame_system::Pallet`
+		/// - Coretime-parachains that:
```

Comment:
```suggestion
		/// - On-demand parachains that:
```
Here the distinction might not be clear between coretime vs PLO. Maybe better as something a bit more descriptive instead "Parachains making blocks on-demand vs parachains making blocks continuously at a fixed period" or something